runtime/dart_isolate.cc

```diff
@@ -29,6 +29,80 @@
 
 namespace flutter {
 
+namespace {
+
+class DartErrorString {
+ public:
+  ~DartErrorString() {
+    if (str_) {
+      ::free(str_);
+    }
+  }
+  char** error() { return &str_; }
+  const char* str() const { return str_; }
+  operator bool() const { return str_ != nullptr; }
+
+ private:
+  char* str_ = nullptr;
```

Comment:
Nit: Disallow copy and assign. This is trivially copyable you might run into situations where a double free happens if this is copied.